shared/teams/role-picker/controlled-container.js

```diff
@@ -0,0 +1,65 @@
+// @flow
+import React from 'react'
+import {RoleOptions} from '.'
+import {connect} from 'react-redux'
+import {compose, withHandlers, withState} from 'recompose'
+import {PopupDialog, HeaderHoc, ScrollView} from '../../common-adapters/index'
+import {isMobile} from '../../constants/platform'
+import {type TypedState} from '../../constants/reducer'
+import {type TeamRoleType} from '../../constants/teams'
+
+/* 
+  Pass through via routeprops
+  onComplete gets selected role
+  selectedRole sets a default role for the component
+  allowOwner specifies whether the user can choose the 'owner' option
+*/
+export type ControlledRolePickerProps = {
+  onComplete: (role: TeamRoleType) => void,
+  selectedRole?: string,
```

Comment:
`TeamRoleType` instead of `string` work?

Reply:
:+1: agreed, this connector can use some typing work